ground/src/main/java/com/google/android/ground/model/geometry/Polygon.kt

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.android.ground.model.geometry
+
+import com.google.common.collect.ImmutableList
+
+class PolygonExteriorHoleException(val shellMaximum: Coordinate, val holeMaximum: Coordinate) :
```

Comment:
Will callers ever need to access the shell min and hole max programmatically? If not, can instead maintain a single GeometryException whose message provides the necessary details for a human to debug the source of the problem?

Reply:
I was trying to model the different particular errors, per Geometry type, as separate exception types all subclassing the same parent--do you think we don't need that? It seems like it'd be helpful to encode this when dealing with different types of Geometries at the same time.

Reply:
If there's some difference in error state that we need/want compile-time checks for then yes, but if it's only to pass additional information about the type of error to the logs, then the additional types are redundant wrt the exception messages.

Reply:
Gotcha, I'll remove these for now then, maybe we're fine with a single exception check.

Reply:
Thanks!